app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,20 +1,28 @@
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def github_auto_sign_up(github_token)
+    return if Errbit::Config.github_org_id.nil?
+
+    # See if the user is a member of the organization that we have access for
+    # If they are, automatically create an account
+    client = Octokit::Client.new(access_token: github_token)
+    client.api_endpoint = Errbit::Config.github_api_url
+    org_ids = client.organizations.map(&:id)
+    return nil unless org_ids.include?(Errbit::Config.github_org_id)
+
+    user_email = github_get_user_email(client)
+    if user_email.nil?
+      flash[:error] = "Could not retrieve user's email from GitHub"
+      nil
+    else
+      User.create(name: env["omniauth.auth"].extra.raw_info.name, email: user_email)
```

Comment:
Doesn't this duplicate part of what was just done in the `github_get_user_email` method?

Reply:
I don't see the duplication, github_get_user_email gets the email.
Then, here we create the user using the email and adding the username, where's the duplication?